IdentityCore/src/webview/embeddedWebview/MSIDOAuth2WebviewController.h

```diff
@@ -26,8 +26,13 @@
 //------------------------------------------------------------------------------
 
 #import <Foundation/Foundation.h>
-#import "MSIDWebViewInteracting.h"
+#import "MSIDWebviewInteracting.h"
+#import "MSIDWebviewDelegate.h"
 
-@interface MSIDOAuth2WebviewController : NSObject <MSIDWebviewInteracting>
+@class MSIDEmbeddedWebviewRequest;
+
+@interface MSIDOAuth2WebviewController : NSObject <MSIDWebviewInteracting, MSIDWebviewDelegate>
+
+- (id)initWithRequest:(MSIDEmbeddedWebviewRequest *)request;
```

Comment:
Let's remove MSIDEmbeddedWebviewRequest class as the logic would be handled in the factory,
So, in here, you can be more clear and do something like
initWithStartURL:endURL:headers:context: etc.

Reply:
addressed